File: functional-tests/src/main/java/com/redhat/che/selenium/core/RhCheLoginPage.java
/*
 * Copyright (c) 2016-2018 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */
package com.redhat.che.selenium.core;

import static java.util.Arrays.asList;

import com.google.inject.Inject;
import org.eclipse.che.selenium.core.SeleniumWebDriver;
import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
import org.eclipse.che.selenium.pageobject.site.LoginPage;
import org.openqa.selenium.TimeoutException;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/** @author Katerina Foniok */
public class RhCheLoginPage implements LoginPage {

  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;

  final private String USERNAME_INPUT_NAME = "username";
  final private String PASSWORD_INPUT_NAME = "password";
  final private String LOGIN_BUTTON_NAME = "login";
  final private String NEXT_BUTTON_ID = "login-show-step2";

  @FindBy(name = USERNAME_INPUT_NAME)
  private WebElement usernameInput;

  @FindBy(name = PASSWORD_INPUT_NAME)
  private WebElement passwordInput;

  @FindBy(name = LOGIN_BUTTON_NAME)
  private WebElement loginButton;

  @FindBy(id = NEXT_BUTTON_ID)
  private WebElement nextButton;

  @Inject
  public RhCheLoginPage(
      SeleniumWebDriver seleniumWebDriver, SeleniumWebDriverHelper seleniumWebDriverHelper) {
    this.seleniumWebDriverHelper = seleniumWebDriverHelper;

    PageFactory.initElements(seleniumWebDriver, this);
  }

  @Override
  public void login(String username, String password) {
    waitOnOpen();

    seleniumWebDriverHelper.setValue(usernameInput, username);
    seleniumWebDriverHelper.waitAndClick(nextButton);
    seleniumWebDriverHelper.setValue(passwordInput, password);
    seleniumWebDriverHelper.waitAndClick(loginButton);

    waitOnClose();
  }

  @Override
  public boolean isOpened() {
    try {
      waitOnOpen();
    } catch (TimeoutException e) {
      return false;
    }

    return true;
  }

  private void waitOnOpen() {
    seleniumWebDriverHelper.waitAllVisibility(asList(usernameInput, nextButton));
  }

  private void waitOnClose() {
    seleniumWebDriverHelper.waitAllInvisibility(asList(passwordInput, loginButton));
  }
}

File: e2e-saas/pageobjects/RhCheLoginPage.ts
/*********************************************************************
 * Copyright (c) 2019 Red Hat, Inc.
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 **********************************************************************/
import 'reflect-metadata';
import { ILoginPage, TYPES, CLASSES, IDriver, TestConstants, DriverHelper } from 'e2e';
import { injectable, inject } from 'inversify';
import { ThenableWebDriver } from 'selenium-webdriver';
import { By } from 'selenium-webdriver';

const USERNAME_INPUT_ID : string = 'username';
const PASSWORD_INPUT_ID : string = 'password';
const NEXT_BUTTON_ID : string = 'login-show-step2';
const LOGIN_BUTTON_ID : string = 'kc-login';

@injectable()
export class RhCheLoginPage implements ILoginPage {
    constructor(
        @inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper,
        @inject(TYPES.Driver) private readonly driver: IDriver) { }

    async login(timeout: number = 20) {
        const webDriver: ThenableWebDriver = this.driver.get();
        await webDriver.navigate().to(TestConstants.TS_SELENIUM_BASE_URL);

        const usernameFieldLocator: By = By.id(USERNAME_INPUT_ID);
        await this.driverHelper.waitVisibility(usernameFieldLocator, 20000);
        await this.driverHelper.enterValue(usernameFieldLocator, TestConstants.TS_SELENIUM_USERNAME, timeout );

        const nextButtonLocator: By = By.id(NEXT_BUTTON_ID);
        await this.driverHelper.waitAndClick(nextButtonLocator, timeout);

        const passwordFieldLocator: By = By.id(PASSWORD_INPUT_ID);
        await this.driverHelper.waitVisibility(passwordFieldLocator, 3000);
        await this.driverHelper.enterValue(passwordFieldLocator, TestConstants.TS_SELENIUM_PASSWORD, timeout );

        const loginButtonLocator: By = By.id(LOGIN_BUTTON_ID);
        await this.driverHelper.waitAndClick(loginButtonLocator, timeout);
    }
}
